src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -504,6 +593,29 @@ DataModel::CommandEntry CodegenDataModelProvider::FirstAcceptedCommand(const Con
 
 DataModel::CommandEntry CodegenDataModelProvider::NextAcceptedCommand(const ConcreteCommandPath & before)
 {
+    CommandHandlerInterface * interface =
+        CommandHandlerInterfaceRegistry::Instance().GetCommandHandler(before.mEndpointId, before.mClusterId);
+
+    if (interface != nullptr)
+    {
+        EnumeratorCommandFinder finder(EnumeratorCommandFinder::Operation::FindNext, before.mCommandId);
+        CHIP_ERROR err = interface->EnumerateAcceptedCommands(before, finder.Callback(), &finder);
+
+        if (err != CHIP_ERROR_NOT_IMPLEMENTED)
+        {
+            auto beforeId = finder.GetFound();
+
+            if ((err == CHIP_NO_ERROR) && beforeId.has_value())
+            {
+                DataModel::CommandEntry result;
+                result.path = ConcreteCommandPath(before.mEndpointId, before.mClusterId, *beforeId);
+                return result;
+            }
+
+            return DataModel::CommandEntry::kInvalid;
+        }
```

Comment:
Would it make sense to just do something like this?

```suggestion
        if (err != CHIP_ERROR_NOT_IMPLEMENTED && err != CHIP_NO_ERROR)
        {
                    return DataModel::CommandEntry::kInvalid;
        }
        auto beforeId = finder.GetFound();
        if ((err == CHIP_NO_ERROR) && beforeId.has_value())
        {
            DataModel::CommandEntry result;
            result.path = ConcreteCommandPath(before.mEndpointId, before.mClusterId, *beforeId);
            return result;
        }
```

Reply:
I updated the code flow to make it shorter at call sites. Still a lot of arguments, but hoping this is easier to read.